base/channels.jl

```diff
@@ -417,9 +425,12 @@ immediately, does not block.
 For unbuffered channels returns `true` if there are tasks waiting
 on a [`put!`](@ref).
 """
-isready(c::Channel) = n_avail(c) > 0
-n_avail(c::Channel) = isbuffered(c) ? length(c.data) : length(c.cond_put.waitq)
-isempty(c::Channel) = isbuffered(c) ? isempty(c.data) : isempty(c.cond_put.waitq)
```

Comment:
I think `length(c.cond_put.waitq)` should also be a non-data-race-y thing to access. Probably not currently implemented as such, but perhaps ideally would be? (rather tricky with the knowledge that it might get removed from the queue and put in another queue, so perhaps not possible though)